src/compiler/diagnosticMessages.json

```diff
@@ -3619,7 +3615,7 @@
         "category": "Error",
         "code": 2846
     },
-    "A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.": {
+    "A function with this return type needs to explicitly return a value.": {

```

Comment:
"must explicitly"? (Not sure what form we use most often)

Reply:
Let's pretend we're starting from scratch. What would you prefer to see as a user?

Reply:
I like this diagnostic, but I might be a bit confused when the return type is inferred. That's unrelated to the message per se though.

Reply:
Can we do something like `A function with return type 'X' needs to explicitly return a value.`? I think this would help with Maria's point about when the return type is inferred.